book/from_flat_php_to_symfony2.rst

```diff
@@ -254,9 +254,11 @@ an individual blog result based on a given id::
     function get_post_by_id($id)
     {
         $link = open_database_connection();
-        $id = intval($id);
-        $result = $link->query('SELECT created_at, title, body FROM post WHERE id = '.$id);
-        $row = $result->fetch(PDO::FETCH_ASSOC);
+        $query = 'SELECT created_at, title, body FROM post WHERE  id=:id';
+        $statement = $pdo->prepare($query);
```

Comment:
sorry my mistake - I always use $pdo
yes, $link fits with existing code

regards,

.. matt ..


On Jun 21, 2016, at 07:57 AM, Christian Flothmann <notifications@github.com> wrote:

In book/from_flat_php_to_symfony2.rst:
> @@ -254,9 +254,11 @@ an individual blog result based on a given id::
>      function get_post_by_id($id)
>      {
>          $link = open_database_connection();
> -        $id = intval($id);
> -        $result = $link->query('SELECT created_at, title, body FROM post WHERE id = '.$id);
> -        $row = $result->fetch(PDO::FETCH_ASSOC);
> +        $query = 'SELECT created_at, title, body FROM post WHERE  id=:id';
> +        $statement = $pdo->prepare($query);
$pdo must be $link here (or you need to rename the variable above).
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.